### ISSUE TITLE
Version 3.0 breaks compatibility with DecSync plugin

### ISSUE THREAD

**Issue by tx1683** [Kozea/Radicale#1071]
Hello.
Changes in version 3.0 and above breaks DecSync storage plugin for Radicale.
https://github.com/39aldo39/Radicale-DecSync/issues/3
Can you provide some guides how to update that plugin to match new versions?

**Comment by jonassmedegaard** [comment]
Quoting tx1683 (2020-06-14 17:15:15)
> Changes in version 3.0 and above breaks DecSync storage plugin for Radicale.
> https://github.com/39aldo39/Radicale-DecSync/issues/3
> Can you provide some guides how to update that plugin to match new versions?

Please consult plugin authors for use/update of their plugins

**Comment by tx1683** [comment]
@jonassmedegaard let me rephrase. Is there any documentation which explains how to modify 2.x plugins, so they will be 3.x compatible?

**Comment by MaMiMa81** [comment]
@jonassmedegaard 
```
Quoting tx1683 (2020-06-14 17:15:15)
 Changes in version 3.0 and above breaks DecSync storage plugin for Radicale.
 39aldo39/Radicale-DecSync#3
 Can you provide some guides how to update that plugin to match new versions?
Please consult plugin authors for use/update of their plugins
```
The  plugin author of Radicale Decsync has indeed now updated is plugin

I have just upgraded Radicale to 3.0.3 & Radicale Decsync to 1.1.0.

On performing the prescribed:

radicale --verify-storage

I get the output:

[2020-06-27 15:19:46 +0100] [7006] [ERROR] Invalid collection '': discover() takes from 2 to 3 positional arguments but 4 were given [2020-06-27 15:19:46 +0100] [7006] [CRITICAL] Storage verifcation failed

However upon testing syncing appears to be performing just as satisfactory as with Radicale to 2.1.1 & Radicale DecSync 1.0.3.

I haven't adjusted any config files during the upgrade & in the Radicale/NEWS.md (changelog) it states:

    File system storage

The storage format is compatible with Radicale 2.x.x
Run radicale --verify-storage to check for errors

So I'm unclear what this means - I have posted over at [https://github.com/39aldo39/Radicale-DecSync/issues/3](url) too.

**Comment by Unrud** [comment]
> > The storage format is compatible with Radicale 2.x.x

This refers to the storage format of the default `multifilesystem` plugin.

> On performing the prescribed:
>
> `radicale --verify-storage
>
> I get the output:
>
> ``[2020-06-27 15:19:46 +0100] [7006] [ERROR] Invalid collection '': discover() takes from 2 to 3 positional arguments but 4 were given [2020-06-27 15:19:46 +0100] [7006] [CRITICAL] Storage verifcation failed`

That's a bug in the Decsync plugin.

**Comment by MaMiMa81** [comment]
The plugin author of Radicale Decsync has indeed now updated his plugin to release v1.1.1 & my issue:
[https://github.com/Kozea/Radicale/issues/1071#issuecomment-650575069](url)
is resolved
Cheers

**Comment by MaMiMa81** [comment]
@tx1683 
You may wish to close this issue now
